pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -1776,13 +1831,17 @@ scheduler_pending_pods{queue="unschedulable"} 20
 			defer cancel()
 
 			m := map[string][]framework.PreEnqueuePlugin{"": {&preEnqueuePlugin{allowlists: []string{queueable}}}}
-			queue := NewTestQueue(ctx, newDefaultQueueSort(), WithClock(testingclock.NewFakeClock(timestamp)), WithPreEnqueuePluginMap(m))
+			recorder := metrics.NewMetricsAsyncRecorder(3, 20*time.Microsecond, ctx.Done())
+			queue := NewTestQueue(ctx, newDefaultQueueSort(), WithClock(testingclock.NewFakeClock(timestamp)), WithPreEnqueuePluginMap(m), WithPluginMetricsSamplePercent(test.pluginMetricsSamplePercent), WithMetricsRecorder(*recorder))
 			for i, op := range test.operations {
 				for _, pInfo := range test.operands[i] {
 					op(queue, pInfo)
 				}
 			}
 
+			// we need to wait for the metrics async recorder to record the metrics.
+			time.Sleep(50 * time.Millisecond)
```

Comment:
Can we eliminate this sleep? e.g., by stopping the recorder via `cancel()`? or make the fake clock steps forward a couple of seconds?

```go
fakeClock := testingclock.NewFakeClock(timestamp)
queue := NewTestQueue(ctx, newDefaultQueueSort(), WithClock(fakeClock), WithPreEnqueuePluginMap(m), WithPluginMetricsSamplePercent(test.pluginMetricsSamplePercent), WithMetricsRecorder(*recorder))
......
fakeClock.Step(5 * time.Second)
```

Reply:
just realize we can call `FlushMetrics` directly. I'll change here to call it and remove sleep here.

Reply:
SG. Thanks.

Reply:
Kindly ping, just in case we complete it before code freeze (Mar 14, 17:00 PDT). 🙏

Reply:
Oops, I forgot to push the change. Wait a minute.

Reply:
add change on here (L1842) + squashed.